src/PostgREST/App.hs

```diff
@@ -206,7 +204,9 @@ app dbStructure conf apiRequest =
         else return notFound
 
     (ActionRead, TargetRoot, Nothing) -> do
-      body <- encode <$> H.query schema accessibleTables
+      body <- if contentType == OpenAPI
+                 then (encodeApi . toTableInfo) <$> H.query schema accessibleTables
+                 else encode <$> H.query schema accessibleTables
       return $ responseLBS status200 [jsonH] $ cs body
```

Comment:
This is kind of pedantic, but if when the swagger is returned then it should respond with a `Content-Type: application/openapi+json` header.